extensions/wikia/AdMixExperiment/scripts/AdMixExperiment.js

```diff
@@ -0,0 +1,294 @@
+
+		function getFirstAdTopPosition() {
+			return $rail.offset().top - globalNavigationHeight;
+		}
+
+		function getFirstAdBottomPosition(firstAdTopPosition) {
+			return firstAdTopPosition + gapSize;
+		}
+
+		function getAdAndRecircWrapperTopPosition() {
+			var visibleElementBeforeWrapperBottom = parseInt($visibleElementBeforeWrapper.offset().top, 10) +
+				visibleElementBeforeWrapperHeight;
+
+			if (recircEnabled) {
+				return visibleElementBeforeWrapperBottom - globalNavigationHeight;
+			} else {
+				return visibleElementBeforeWrapperBottom + noRecircAdBottomSpace -
+					(viewportHeight - $adAndRecircWrapper.outerHeight(true));
+			}
+		}
+
+		function getAdAndRecircWrapperBottomPosition(adAndRecircWrapperTopPosition) {
+			return adAndRecircWrapperTopPosition + gapSize;
+		}
+
+		function getStopPosition() {
+			var footerOffsetTop = parseInt($footer.offset().top, 10);
+			var adMixRecircWrapperHeight = $adAndRecircWrapper.outerHeight(true);
+
+			if (recircEnabled) {
+				return footerOffsetTop - globalNavigationHeight - adMixRecircWrapperHeight - bottomMargin;
+			} else {
+				return footerOffsetTop - globalNavigationHeight - adMixRecircWrapperHeight -
+					noRecircAdBottomSpace - magicNumber;
+			}
+		}
+
+		function reset() {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			resetRecircStyles();
+		}
+
+		function scrollAfterAdAndRecircWrapperTopPosition() {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', gapSize + 'px');
+
+				$recirc.css({
+					position: 'fixed',
+					top: globalNavigationHeight + 'px',
+					width: '280px'
+				});
+			}
+
+			resetAdAndRecircWrapper();
+		}
+
+		function scrollAfterAdAndRecircWrapperBottomPosition() {
+			var adAndRecircWrapperStyles = {
+				position: 'fixed',
+				width: '300px'
+			};
+
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			if (recircEnabled) {
+				$recircWrapper.css('margin-bottom', '0');
+
+				$recirc.css({
+					position: '',
+					top: '',
+					width: ''
+				});
+
+				adAndRecircWrapperStyles.top = globalNavigationHeight + 'px';
+			} else {
+				adAndRecircWrapperStyles.bottom = noRecircAdBottomSpace + 'px';
+				adAndRecircWrapperStyles.top = '';
+			}
+
+			$adAndRecircWrapper.css(adAndRecircWrapperStyles);
+		}
+
+		function scrollAfterFirstAdTopPosition() {
+			resetRecircStyles();
+			resetTopAdPaddings();
+			$topRightAd.css({
+				position: 'fixed',
+				top: globalNavigationHeight + 'px',
+				width: '300px'
+			});
+		}
+
+		function scrollAfterFirstAdBottomPosition() {
+			resetRecircStyles();
+			resetTopAd();
+			$topRightAdWrapper.css({
+				'padding-bottom': '',
+				'padding-top': gapSize + 'px'
+			});
+		}
+
+		function scrollStopPosition(stopPosition) {
+			if (topRightAdFixed) {
+				resetTopAd();
+				resetTopAdPaddings();
+			}
+
+			// FIXME cache it? at least the element
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Take the required action to fix the issue indicated by this comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)